### PR TITLE
Fix kustomize config to conform to kubebuilder v2 layout

### LINKED ISSUE #405: Migrate to Kubebuilder v2
/kind feature

**Describe the solution you'd like**
Update controller to [Kubebuilder V2](https://book.kubebuilder.io/), Some advantages:

 * Use of Go Modules
 * Integration with cert-manager for webook certificates
 * Handling multiple API versions
 * Ability to use Roles rather than ClusterRoles

Solves #404 

**Anything else you would like to add:**

I have done the migration once and is well documented by Kubebuilder though would a big code change due to restructured code paths and methods.

### PR DESCRIPTION
**Pull request by yuzisun**
**What this PR does / why we need it**:
- Fix current kustomize config layout
- Remove status/address patch
- add `maxDescLen=0` to remove description string to reduce crd size per https://github.com/kubernetes-sigs/kubebuilder/issues/1140

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #405 

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/testing/envtest_setup.go [hunk 1]
```diff
@@ -38,9 +38,9 @@ func SetupEnvTest() *envtest.Environment {
 		// The relative paths must be provided for each level of test nesting
 		// This code should be illegal
 		CRDDirectoryPaths: []string{
-			filepath.Join("..", "..", "..", "..", "..", "config", "default", "crds", "base"),
+			filepath.Join("..", "..", "..", "..", "..", "config", "crd"),
```
Comment (ellistarn) [inline pkg/testing/envtest_setup.go:41]: Do we still need both of these paths?

Comment (yuzisun) [reply pkg/testing/envtest_setup.go:41]: yes, we have tests at different levels
`pkg/controller/inferenceservice/reconcilers/knative`
`pkg/controller/inferenceservice`

**Pull request merged**